File: easy/python/Question3090.py
from collections import defaultdict


class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        ans = left = 0
        cnt = defaultdict(int)
        for right, ch in enumerate(s):
            cnt[ch] += 1
            while cnt[ch] > 2:
                cnt[s[left]] -= 1
                left += 1
            ans = max(ans, right - left + 1)
        return ans


print(Solution().maximumLengthSubstring("aaaa"))

File: easy/cpp/Question3090.cpp
#include <bits/stdc++.h>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int maximumLengthSubstring(string s) {
        unordered_map<int, int> map;
        int max_len = 0, left = 0;
        for (int right = 0; right < s.length(); right++) {
            while (map[s[right]] == 2) {
                map[s[left]]--;
                left++;
            }
            map[s[right]]++;
            max_len = max(max_len, right - left + 1);
        }
        return max_len;
    }
};
